### PR TITLE
Upgrade apache commons-logging to current version (1.1.3 -> 1.2)

### PR DESCRIPTION
**Pull request by pgier**
This is a bug fix upgrade and removes support for very old JDK versions (<1.2).
More information here: https://commons.apache.org/proper/commons-logging/changes-report.html

### CODE REVIEW

File: plugins/repository-hdfs/build.gradle [hunk 1]
```diff
@@ -44,7 +44,7 @@ dependencies {
   compile 'org.apache.htrace:htrace-core:3.1.0-incubating'
   compile 'com.google.guava:guava:11.0.2'
   compile 'com.google.protobuf:protobuf-java:2.5.0'
-  compile 'commons-logging:commons-logging:1.1.3'
+  compile 'commons-logging:commons-logging:1.2'
```
Comment (jasontedor) [inline plugins/repository-hdfs/build.gradle]: No we can't do this one, the version of Hadoop here depends on [1.1.3](https://github.com/apache/hadoop/blob/release-2.7.1/hadoop-project/pom.xml#L591-L610).

File: buildSrc/version.properties [hunk 2]
```diff
@@ -20,7 +20,7 @@ httpclient        = 4.5.2
 httpcore          = 4.4.5
 # When updating httpasyncclient, please also update core/src/main/resources/org/elasticsearch/bootstrap/test-framework.policy
 httpasyncclient   = 4.1.2
-commonslogging    = 1.1.3
+commonslogging    = 1.2
```
Comment (jasontedor) [inline buildSrc/version.properties]: I think we need to look at what modules have a dependency on commonslogging and see if we can change it across the board like this. This also impacts downstream projects like plugins that rely on the version properties here in core to determine the version that they use.

**Review by jasontedor — CHANGES_REQUESTED** [review]
The Hadoop dependency definitely can not change, and I think that there is investigation needed before we can safely make the larger change.

File: buildSrc/version.properties (hunk 2 above)
Comment (jasontedor) [inline buildSrc/version.properties]: I look at all the uses of commonslogging, and I think that going to 1.2 is safe here with the exception of the Hadoop change which as I mentioned must stay on 1.1.3.

**Comment by jasontedor** [comment]
Can you please revert the repository-hdfs change and run `gradle updateSHAs` and push the resulting sha changes?

**Comment by jasontedor** [comment]
test this please

**Comment by pgier** [comment]
Thanks for the quick feedback!  I have pushed the requested changes.

**Comment by jasontedor** [comment]
> Thanks for the quick feedback! I have pushed the requested changes.

You're welcome. Thanks for the PR and the quick iteration. I have asked our CI to kick of a build.

**Comment by jasontedor** [comment]
I think that the transitive commons-logging dependency in repository-hdfs needs to be excluded.

**Comment by rjernst** [comment]
@jasontedor I'm not sure that will work. When we moved repository-hdfs into ES, we did so adding each dependency (ie starting with excluding all transitive deps, and adding them one by one) until it worked. So I don't think simply dropping commons-logging will work.

**Comment by jasontedor** [comment]
I meant the conflicting transitive dependency only but keeping the explicitly defined one.

**Pull request closed without merging**